shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
> Apparently wrapping the capability with new Capnp.Capability() before returning it causes this close() to be shadowed or something. If I remove that wrapper, close() gets called as expected.

I don't understand. How does the code work at all if you don't covert the object to a capability?

> Moreover, I think this is the wrong behavior for close().

I see what you're getting at here, but conversely, the logic in `dropInternal()` is arguably wrong too: it should not dismiss the notification if any live capabilities still exist. But, detecting live capabilities is hard. Since it doesn't detect live capabilities, if `close()` were implemented as you suggest, then the `dismissNotification()` call could end up happening twice, if the capability were saved and then dropped before the live capability was closed. This seemed to me to be worse than the current behavior.